content/documentation/pages/3-stream-developer-guides/5-troubleshooting/3-debugging-scdf.md

```diff
@@ -4,7 +4,134 @@ title: 'Debugging Stream applications deployed by Data Flow'
 ---
 
-A Stream is deployed and the status isn't 'deployed'. Now what?
-How to debug when using SCDF.
+# Debugging streams
 
-[Here is an example](https://stackoverflow.com/questions/55612650/cant-change-content-type-from-application-octet-stream-to-application-json-in-r)
+If a stream fails to deploy:
+
+- Ensure the latest GA of a particular release version is being used
+- Ensure platform of choice meets at least the minimum supported version
+- All applications that are part of the stream have been tested as standalone applications outside of SCDF
+
+Skipper is responsible for deploying streams.
+Stream deployment failure messages can typically be found in the Skipper application logs.
+
+# Local
+
+```mermaid
+graph TD;
+    A{Start Locally?} --> |No| A
+    A --> |Yes| B(Exceptions in SCDF or Skipper logs?)
+	B --> |Yes| C(Resolve exceptions)
+	B --> |No| D(Exceptions in app logs?)
+	D -->|Yes| E(Resolve exceptions)
+	D -->|No| F(Raise log levels)
+    F --> G(Enable debug)
+```
+
+Application log files can be inspected on a per application basis.
+To aggregate logs from all applications into one, the deployer property `inheritLogging=true` can be set.
+See
+[Log Redirect](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#_log_redirect)
+for more information and [Deployment Logs](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#troubleshooting-deployment-logs) for enabling more log output.
+
+Debugging applications via JDWP can be accomplished by setting the deployer property `debugPort`.
+See [Remote Debugging](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#_remote_debugging) for more information.
+
+# Docker Compose - Startup
+
+```mermaid
+graph TD;
+    A{DATAFLOW_VERSION defined?} --> |No| A
+    A --> |Yes| B{SKIPPER_VERSION defined?}
+	B --> |No| B
+```
+
+The environment variables `DATAFLOW_VERSION` and `SKIPPER_VERSION` must be available in the current shell environment via `export` or prefixing the `docker-compose` command.
+See [Starting Docker Compose](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#getting-started-local-deploying-spring-cloud-dataflow-docker-starting) for more information.
+
+# Docker Compose - Runtime
+
+```mermaid
+graph TD;
+    A{Low Resources?} --> |Yes| B(Increase Docker Resources)
+    A --> |No| C{Exceptions in app logs?}
+	C --> |Yes| D(Resolve exceptions)
+```
+
+By default, the amount of memory allocated to Docker may be too low.
+The command `docker stats` can provide useful information into resource usage.
+If applications are failing to deploy due to resource constraints, increase resource allocations.
+Consult the [Docker documentation](https://docs.docker.com/) for your platform.
+
+As streams are deployed via Skipper, applications that are part of that stream will be launched as Java processes on the Skipper container.
```

Comment:
Perhaps also add a note about "the more the number of streams, the more memory is required for skipper container to run the apps in them" or something like that. 

See https://github.com/spring-cloud/spring-cloud-dataflow/issues/3107#issuecomment-477250327 for background.

Reply:
updated